File: sopaDeLetras.java
import java.util.*;
import java.util.Random;


class Main {
  
    public class GeneradorAleatorio {
        public static char[][] generarSopa(int filas, int columnas, int rango) {
            Random random = new Random();
            char[][] sopaDeLetras = new char[filas][columnas];

            for (int i = 0; i < filas; i++) {
                for (int j = 0; j < columnas; j++) {
                    char caracter = (char) ('a' + random.nextInt(rango));
                    sopaDeLetras[i][j] = caracter;
                }
            }

            return sopaDeLetras;
        }
    }
    
    public static void main(String[] args) {
        int fila = 8;
        int columna = 8;
        int rango = 26;

        char palabraGenerada[][] = GeneradorAleatorio.generarSopa(fila, columna, rango);
        System.out.println("Sopa De Letras");
        for(int i = 0;i < fila; i++){
            for(int j=0; j <columna;j++){
                System.out.print(palabraGenerada[i][j] + " ");
            }
            System.out.print("\n");
        }
    }
}

File: sopaDeLetras.cpp
#include <iostream>
#include <vector>
#include <cstdlib>
#include <ctime>
using namespace std;

vector<vector<char>> generarSopa(int filas, int columnas, int rango) {
    vector<vector<char>> sopaDeLetras(filas,vector<char>(columnas));
    srand(time(0));

    for (int i = 0; i < filas; ++i) {
        for (int j = 0; j < columnas; ++j) {
            char caracter = 'a' + rand() % rango;
            sopaDeLetras[i][j] = caracter;
        }
    }

    return sopaDeLetras;
}

int main() {
    const int filas = 8;
    const int columnas = 8;
    const int rango = 26;

    vector<vector<char>> sopa = generarSopa(filas, columnas, rango);


    for (const auto& fila : sopa) {
        for (char c : fila) {
            cout << c;
        }
        cout << '\n';
    }

    return 0;
}